firebase-appdistribution/src/main/java/com/google/firebase/appdistribution/impl/ImageUtils.java

```diff
@@ -0,0 +1,67 @@
+package com.google.firebase.appdistribution.impl;
+
+import android.graphics.Bitmap;
+import android.graphics.BitmapFactory;
+import androidx.annotation.Nullable;
+import com.google.auto.value.AutoValue;
+import java.io.File;
+
+public class ImageUtils {
+
+  @AutoValue
+  abstract static class ImageSize {
+    abstract int width();
+
+    abstract int height();
+  }
+
+  /**
+   * Read an image from a file, scaled as small as possible according to the target size.
+   *
+   * <p>The returned bitmap will be scaled down, preserving the aspect ratio, by the largest power
+   * of 2 that results in the width and height still being larger than the target.
+   *
+   * <p>Based on https://developer.android.com/topic/performance/graphics/load-bitmap#load-bitmap.
+   *
+   * @return the image, or null if it could not be decoded
+   * @throws IllegalArgumentException if target height or width are less than or equal to zero
+   */
+  @Nullable
+  public static Bitmap readScaledImage(File file, int targetWidth, int targetHeight) {
+    if (targetWidth <= 0 || targetHeight <= 0) {
+      throw new IllegalArgumentException(
+          String.format(
+              "Tried to read image with bad dimensions: %dx%d", targetWidth, targetHeight));
+    }
+    ImageSize imageSize = getImageSize(file);
+    final BitmapFactory.Options options = new BitmapFactory.Options();
+    options.inSampleSize =
+        calculateInSampleSize(imageSize.width(), imageSize.height(), targetWidth, targetHeight);
+    return BitmapFactory.decodeFile(file.getAbsolutePath(), options);
+  }
+
+  private static ImageSize getImageSize(File file) {
```

Comment:
nit: This could be moved to ImageSize to act as a factory method. Proposed method name: `read()`